### PR TITLE
feat(react-components): add instance reference component in the floating panel when creating a POI and some other improvements

### PR DESCRIPTION
**Pull request by danpriori**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:
Change the Custom based input command to support more field types in order to add instance reference in floating panel when creating a POI and some improvements in the instance label component

image

## Checklist :ballot_box_with_check:

- [ ] I am proud of this feature.
- [X] I have performed a self-review of my own code.
- [X] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [X] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/base/commands/CustomBaseInputCommand.ts [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-  content: string;
-};
+export type FieldContent =
+  | {
+      type: 'text' | 'commentWithButtons' | 'comment';
+      content: string;
+    }
+  | {
+      type: 'customInput';
+      content: ReactNode;
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/base/commands/CustomBaseInputCommand.ts:15]: Nils Petter is not a fan of his architecture being reliant on React - but let's fix that after release

File: react-components/src/architecture/base/commands/CustomBaseInputCommand.ts [hunk 2]
```diff
@@ -51,7 +61,33 @@ export abstract class CustomBaseInputCommand extends RenderTargetCommand {
...
+        };
+      }
+      if (content.type === 'submitButtons') {
+        return {
+          type: content.type,
+          content: undefined
+        };
+      }
+      return content;
+    });
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/base/commands/CustomBaseInputCommand.ts:90]: Could this be a "clearFieldContent" function?

File: react-components/src/architecture/concrete/pointsOfInterest/types.ts [hunk 3]
```diff
@@ -37,8 +38,8 @@ export function createPointsOfInterestPropertiesFromPointAndTitle(
     positionZ: cdfPosition.z,
     scene,
     sceneState: {},
-    name: contents[0],
-    description: contents[1],
+    name: contents[0] as string,
+    description: contents[1] as string,
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/pointsOfInterest/types.ts:42]: Burns in my eyes :sweat_smile:

**Review by haakonflatval-cognite — APPROVED** [review]
I don't like this code, but let's go!

File: react-components/src/architecture/base/commands/CustomBaseInputCommand.ts (hunk 1 above)
Comment (danpriori) [reply react-components/src/architecture/base/commands/CustomBaseInputCommand.ts:15]: yep

File: react-components/src/architecture/base/commands/CustomBaseInputCommand.ts (hunk 2 above)
Comment (danpriori) [reply react-components/src/architecture/base/commands/CustomBaseInputCommand.ts:90]: yes sure. Let's do it after release. Including some refactoring on this logic, simplifying it or even not using this approach

File: react-components/src/architecture/concrete/pointsOfInterest/types.ts (hunk 3 above)
Comment (danpriori) [reply react-components/src/architecture/concrete/pointsOfInterest/types.ts:42]: It burns my eyes too 🙈but it will be changed definitely after release

**Pull request merged**